src/Events.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core;
+
+/**
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class Events
+{
+    public const PRE_READ = 'api_platform.pre_read';
+    public const READ = 'api_platform.read';
```

Comment:
This event will not be used by GraphQL and is internal. Should we expose it elsewhere? Should we call it something else (`api_platform.rest.read`)?

Reply:
WDYT about keeping this event name, and dispatch a `api_platform.query` event for GraphQL?

Reply:
I don't understand how the `api_platform.read` event could be only internal and not listened by any listener in a project

Reply:
For GraphQL, there will be no event like this dispatched (only pre and post). GraphQL is working differently. Using `query` does not work: there is a read step in the mutation resolver for example.
`api_platform.read` _could_ be listened, but it shouldn't since it is used by our `ReadListener`.